syntax = "proto3";

service RaftService {
    rpc RequestVote(RequestVoteRequest) returns (RequestVoteResponse);
    rpc AppendEntries(AppendEntryRequest) returns (AppendEntriesResponse);
    rpc GetLeader(EmptyRequest) returns (GetLeaderResponse);
    rpc Suspend(SuspendRequest) returns (SuspendResponse);
}

message RequestVoteRequest {
    int32 term = 1;
    int32 candidateId = 2;
}

message RequestVoteResponse {
    int32 term = 1;
    bool result = 2;
}

message AppendEntryRequest {
    int32 term = 1;
    int32 leaaderId = 2;
}

message AppendEntriesResponse {
    int32 term = 1;
    bool success = 2;
}

message EmptyRequest{
}

message SuspendResponse {
    string message = 1;
}

message GetLeaderResponse{
    int32 leaderId = 1;
    string address = 2;
}

message SuspendRequest {
    int32 period = 1;
}

